packages/carbon-graphs/src/js/helpers/axis.js

```diff
@@ -1394,6 +1412,59 @@ const getUpperOutlierStretchFactorList = (config) => {
   return upperStretchFactors;
 };
 
+/**
+ * Calculates the upper part of the outlier based on data points.
+ * If the content has any data points that are outside the lower and upper bounds set
+ * in the vertical axis then we adjust the axis bounds to support that outlier value.
+ *
+ * @private
+ * @param {object} config - config object derived from input JSON
+ * @returns {Array} List of upper bound values for each of the vertical axis
+ */
+const getUpperOutlierStretchFactorXAxis = (config) => {
+  const upperStretchFactors = [];
+  const getMaxValue = (conf, xAxis, axisMaxValue) => {
+    const dataRangeMaxValue = conf.axis[xAxis].dataRange.max;
+    return dataRangeMaxValue > axisMaxValue
+      ? dataRangeMaxValue
+      : axisMaxValue;
+  };
+  const getUpperStretchFactor = (xAxis) => {
+    const axisMaxValue = config.axis[xAxis].domain.upperLimit;
+    const axisMidPoint = getMidPoint(config, xAxis);
+    const upperStretchFactor = Math.abs(
+      (getMaxValue(config, xAxis, axisMaxValue) - axisMidPoint)
+                / (axisMaxValue - axisMidPoint),
+    );
+    return upperStretchFactor > 1 ? upperStretchFactor : 1;
+  };
+  upperStretchFactors.push(getUpperStretchFactor(constants.X_AXIS));
+
+  return upperStretchFactors;
+};
+
+/**
+* Determines if the values provided exceed the lower and upper bounds provided in the Y or Y2 axes
+* If the values exceed the bounds then the range and domain are adjusted accordingly.
+* There is no outlier check for X axis, for now, due to the possibility that X axis can be a timeseries.
+*
+* @private
+* @param {object} config - config object derived from input JSON
+* @returns {object} stretch factor determines the new upper and lower limit.
+*/
+//
+const determineOutlierStretchFactorXAxis = (config) => {
+  const sortOutlier = (firstValue, secondValue) => secondValue - firstValue;
+  return {
+    upperLimit: getUpperOutlierStretchFactorXAxis(config).sort(
+      sortOutlier,
+    )[0],
+    lowerLimit: getLowerOutlierStretchFactorList(config).sort(
```

Comment:
It seems like both `getUpperOutlierStretchFactorXAxis` and `getLowerOutlierStretchFactorList` always returns an array with only one value in it. Is the sorting added in preparation for future changes that they will return multiple values in the array?